tests/unit/amo/components/TestConfirmationDialog.js

```diff
@@ -1,36 +1,20 @@
 import * as React from 'react';
+import { createEvent, fireEvent } from '@testing-library/react';
 
-import Button from 'amo/components/Button';
-import ConfirmationDialog, {
-  ConfirmationDialogBase,
-} from 'amo/components/ConfirmationDialog';
-import {
-  createFakeEvent,
-  fakeI18n,
-  shallowUntilTarget,
-} from 'tests/unit/helpers';
+import ConfirmationDialog from 'amo/components/ConfirmationDialog';
+import { render as defaultRender, screen } from 'tests/unit/helpers';
 
 describe(__filename, () => {
-  function render(otherProps = {}) {
-    const props = {
-      i18n: fakeI18n(),
-      onCancel: sinon.stub(),
-      onConfirm: sinon.stub(),
-      ...otherProps,
-    };
-    return shallowUntilTarget(
-      <ConfirmationDialog {...props} />,
-      ConfirmationDialogBase,
-    );
+  function render(props = {}) {
+    return defaultRender(<ConfirmationDialog {...props} />);
   }
 
   it('renders a dialog', () => {
     const message = 'this action is risky, are you sure?';
-    const root = render({ message });
-
-    expect(root.find('.ConfirmationDialog-message')).toHaveLength(1);
-    expect(root.find('.ConfirmationDialog-message')).toHaveText(message);
+    render({ message });
 
+    expect(screen.getByText(message)).toHaveClass('ConfirmationDialog-message');
+    /*
     expect(root.find(Button)).toHaveLength(2);
```

Comment:
Those commented assertions are testing that the dialog contains the right buttons, in the even looking at the order they appear. Thoughts on how / whether we should test that ?

Reply:
I don't think we need to be testing the order in which they appear. That might not have been the intent of the original test - it might just be a byproduct of the fact that we are accessing them via `find(Button).at(x)`.

Other tests in this file verify that the buttons are displayed (e.g., the tests that click on the Confirm and Cancel buttons), and accessing them via `getByRole` and using `name`, as I suggested in a different comment, also verifies their default text. 

The existing test is also checking that each button has the expected class assigned (e.g., `ConfirmationDialog-confirm-button` for the Confirm button), so, if we want to continue with those assertions we should have a test somewhere for that, but I'm not sure we do. We had/have a philosophy that we do not test for anything hardcoded. For example, if a button always has the text of "Submit", and there was nothing in the code that would allow that to be changed, then we wouldn't add an assertion that the button has the text "Submit". As these classes are hardcoded, it could be argued that we don't need assertions for them. One could pose the opposite argument that although they are hardcoded, they represent an important piece of code, as they instruct the browser how to style the element, and if someone was to change one accidentally that could result in a regression. So really, I think we could either test them or not. We might want to make a call about that at this point and include it in our best practices docs.

One thing that is commented out and doesn't seem to be covered in any existing tests are assertions about the default `buttonType` for each button. We should keep those assertions. Where to place them (in this test, in another test, or in a new test) is really up to you.

Reply:
The other thing I wasn't sure about was whether or not to replicate the fact that this test was expecting the buttons to be inside the component in the DOM. I see we have the `within` helper for that, any reason not to use it here ?

Reply:
I think we should only use `within` when there is an absolute need for it. If it is important to isolate something that could appear in the document multiple times, we can use `within` to target a particular instance, but in this case I don't see any reason to use it. We don't want to be testing layout, we want to be testing functionality.